### PR TITLE
CIS: DNS Record

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

**Comment by kavya498** [comment]
Reduce the number of commits, preferably to two..
one for the resource changes
other for vendor changes

File: ibm/resource_ibm_cis_dns_record.go [hunk 1]
```diff
@@ -444,52 +569,453 @@ func resourceIBMCISDnsRecordUpdate(d *schema.ResourceData, meta interface{}) err
...
+	}
+
+	d.Set(cisDNSRecordID, *result.Result.ID)
+	d.Set(cisZoneName, *result.Result.ZoneName)
+	d.Set(cisDNSRecordCreatedOn, *result.Result.CreatedOn)
+	d.Set(cisDNSRecordModifiedOn, *result.Result.ModifiedOn)
+	d.Set(cisDNSRecordName, *result.Result.Name)
+	d.Set(cisDNSRecordType, *result.Result.Type)
+	d.Set(cisDNSRecordContent, *result.Result.Content)
+	d.Set(cisDomainID, *result.Result.ZoneID)
```
Comment (kavya498) [inline ibm/resource_ibm_cis_dns_record.go:633]: running terraform apply twice returns infrastructure change..
please set domainID as combination of <zoneID>:<cisID>

File: ibm/data_source_ibm_cis_dns_record.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		record[cisDNSRecordTTL] = *instance.TTL
+
+		switch *instance.Type {
+		// for MX & SRV records ouptut
+		case cisDNSRecordTypeMX, cisDNSRecordTypeSRV:
+			record[cisDNSRecordPriority] = *instance.Priority
+		}
+		records = append(records, record)
+	}
+	d.SetId(dataSourceIBMCISDNSRecordID(d))
```
Comment (kavya498) [inline ibm/data_source_ibm_cis_dns_record.go:155]: set id as combination of record_id:zone_id:cis_id
please set required parameters of schema in read method

File: ibm/resource_ibm_cis_dns_record.go (hunk 1 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_dns_record.go:633]: ZoneID:cisID set appropriately as mentioned already.  

I tried to do ```terraform apply``` twice... it does not change anything. Please check the log below,

```bash
kumarganesan@Kumars-MBP$ tfa
ibm_cis_dns_record.test_dns_a_record: Creating...
ibm_cis_dns_record.test_dns_a_record: Creation complete after 5s [id=918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

a_record_output = {
  "cis_id" = "crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "content" = "1.2.3.4"
  "created_on" = "2020-08-24T06:42:15.404107Z"
  "domain_id" = "fd613592752217b573edbf741c893ecb"
  "id" = "918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "modified_on" = "2020-08-24T06:42:15.404107Z"
  "name" = "test-exmple.sdk.cistest-load.com"
  "proxiable" = true
  "proxied" = true
  "record_id" = "918df76aaea20008041073c33244bfa7"
  "ttl" = 1
  "type" = "A"
}

--------------------------------------------------------------------------------------------------------------------------------------------------
[Mon Aug 24 12:12:17 : ~/work/src/github.com/tf/examples/ibm-cis-example]
kumarganesan@Kumars-MBP$ tfa
ibm_cis_dns_record.test_dns_a_record: Refreshing state... [id=918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

a_record_output = {
  "cis_id" = "crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "content" = "1.2.3.4"
  "created_on" = "2020-08-24T06:42:15.404107Z"
  "domain_id" = "fd613592752217b573edbf741c893ecb"
  "id" = "918df76aaea20008041073c33244bfa7:fd613592752217b573edbf741c893ecb:crn:v1:staging:public:internet-svcs-ci:global:a/01652b251c3ae2787110a995d8db0135:f66f8dd7-cf18-49cc-bea8-792071baff2f::"
  "modified_on" = "2020-08-24T06:42:15.404107Z"
  "name" = "test-exmple.sdk.cistest-load.com"
  "proxiable" = true
  "proxied" = true
  "record_id" = "918df76aaea20008041073c33244bfa7"
  "ttl" = 1
  "type" = "A"
}
```

File: ibm/data_source_ibm_cis_dns_record_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			},
+		},
+	})
+}
+
+func testAccCheckIBMNetworkCISDNSRecordsDataSourceConfig(crn, zoneID string) string {
+	// status filter defaults to empty
+	return fmt.Sprintf(`
+	data "ibm_network_cis_dns_records" "test" {
+		crn     = %s
```
Comment (kavya498) [inline ibm/data_source_ibm_cis_dns_record_test.go:35]: names are been passed to IDs, please pass id to config

**Pull request closed without merging**